state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1832,18 +1834,6 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 					source_address,
 				});
 
-				let destination_chain: ForeignChain = destination_asset.into();
-				if deposit_metadata.is_some() && !destination_chain.ccm_support() {
-					Self::deposit_event(Event::<T, I>::DepositFailed {
-						reason: DepositFailedReason::CcmUnsupportedForTargetChain,
-						origin: origin.into(),
-						asset,
-						amount: amount_after_fees,
-						deposit_details,
-						action: Box::new(action),
-					});
-					return DepositAction::NoAction;
-				}
```

Comment:
I moved this code that checks CCM. As I see it we only need this check for vault swaps because for channel swaps this is already checked when opening the deposit channel. Doing this we avoid having to pass unnecessary paramters all the way down to `perform_channel_action`.